Assets/Scripts/Combat/PossessionManager.cs

```diff
@@ -112,17 +117,22 @@ void OnPlayerDeath()
     public void GetPossessed(GameObject playerObj)
     {
         if (isPlayerPossessing.value) return;
-        SetEnemyComponentsEnabled(false);
-        SetPlayerComponentsEnabled(true);
-        Destroy(playerObj);
-        isPossessed = true;
-        isPlayerPossessing.value = true;
-        gameObject.name = PREFIX_WHILE_POSSESSED + initialName;
-        gameObject.tag = Constants.PLAYER_TAG;
-        gameObject.layer = Layer.Parse(LAYER_WHILE_POSSESSED);
-        body.interpolation = RigidbodyInterpolation2D.Interpolate;
-        body.collisionDetectionMode = CollisionDetectionMode2D.Continuous;
-        GlobalEvent.Invoke.OnEnemyPossessed(playerMain);
+        if (canGetPossessed)
+        {
```

Comment:
extremely minor: the early return pattern (like the line above) is one way to keep code cleaner by avoiding nested indentation. Purely coding style / preference though

Reply:
Good call, this is much cleaner